templates.go

```diff
@@ -50,7 +50,7 @@ var (
 	// {{$lis}} is a global in-memory buffer that listeners that is
 	// referenced by the generated mobile APIs, such that all client calls
 	// will be going through it.
-	{{$lis}} = bufconn.Listen(100)
+	{{$lis}} *bufconn.Listener
```

Comment:
To keep this backwards compatible, we could just let this stay the same, then make the method below called `RecreateListeners`. That way it's not mandatory to call it.

Reply:
Perhaps, but IIUC lnd as of right now in master is already incompatible falafel until https://github.com/lightningnetwork/lnd/pull/5010 is merged.

I would suggest we keep it as is and work on getting https://github.com/lightningnetwork/lnd/pull/4599 merged.

Reply:
One possible solution could be to listen for the shutdown signal in a goroutine and then `RecreateListeners` there.

Reply:
> Perhaps, but IIUC lnd as of right now in master is already incompatible with falafel until [lightningnetwork/lnd#5010](https://github.com/lightningnetwork/lnd/pull/5010) is merged.

Sure, but I see no downside on having the listeners be created by default? (without first calleing `NewListeners`)

Reply:
@halseth Yeah it's mostly a question on how and where it would be reinitialized then. I can change it to `RecreateListeners` and then use the shutdown channel for recreation.

Then something like this in lnd's mobile code:

```diff
@@ -134,4 +134,11 @@ func Start(extraArgs string, unlockerReady, rpcReady Callback) {
 
                rpcReady.OnResponse([]byte{})
        }()
+
+       go func() {
+               <-shutdownInterceptor.ShutdownChannel()
+               // When lnd and the gRPC server goes down, it will close our listeners.
+               // So we have to re-initialize them for the next run
+               RecreateListeners()
+       }()
 }
```

Reply:
Ah, my idea was that you call it the same way as you are using `NewListeners` now.

Reply:
@halseth Sure that works too, I guess it costs next to nothing doing that, just a bit ugly. 😛